applications/MeshingApplication/external_includes/trigen_glass_forming.h

```diff
@@ -0,0 +1,1363 @@
+                work_point.X() = xc;
+                work_point.Y() = yc;
+                work_point.Z() = 0.0;
+
+                number_of_points_in_radius = nodes_tree2.SearchInRadius(work_point, radius*1.01, Results.begin(),
+                                             ResultsDistances.begin(),  MaximumNumberOfResults);
+
+                Triangle2D3<Node<3> > geom(
+                    *( (nodes_begin +  in2.trianglelist[base]-1).base() 	),
+                    *( (nodes_begin +  in2.trianglelist[base+1]-1).base() 	),
+                    *( (nodes_begin +  in2.trianglelist[base+2]-1).base() 	)
+                );
+
+                //check if inside and eventually interpolate
+                for( PointIterator it_found = Results.begin(); it_found != Results.begin() + number_of_points_in_radius; it_found++)
+                {
+                    bool is_inside = false;
+                    is_inside = CalculatePosition(x1[0], x1[1],
+                                                  x2[0], x2[1],
+                                                  x3[0], x3[1],
+                                                  (*it_found)->X(),(*it_found)->Y(),N);
+
+
+                    if(is_inside == true)
+                    {
+                        Interpolate(  geom,  N, step_data_size, *(it_found ) );
+
+                    }
+                }
+            }
+        }
+
+        ThisModelPart.Elements().clear();
+
+        //set the coordinates to the original value
+        for( PointVector::iterator it =  list_of_new_nodes.begin(); it!=list_of_new_nodes.end(); it++)
+        {
+            const array_1d<double,3>& disp = (*it)->FastGetSolutionStepValue(DISPLACEMENT);
+            (*it)->X0() = (*it)->X() - disp[0];
+            (*it)->Y0() = (*it)->Y() - disp[1];
+            (*it)->Z0() = 0.0;
+        }
+        //cleaning unnecessary data
+        //in2.deinitialize();
+        //in2.initialize();
+        //free( in2.pointlist );
+
+        //add preserved elements to the kratos
+        Properties::Pointer properties = ThisModelPart.GetMesh().pGetProperties(1);
+        nodes_begin = ThisModelPart.NodesBegin();
+        (ThisModelPart.Elements()).reserve(out2.numberoftriangles);
+
+        for(int iii = 0; iii< out2.numberoftriangles; iii++)
+        {
+            int id = iii + 1;
+            int base = iii * 3;
+            Triangle2D3<Node<3> > geom(
+                *( (nodes_begin +  out2.trianglelist[base]-1).base() 	),
+                *( (nodes_begin +  out2.trianglelist[base+1]-1).base() 	),
+                *( (nodes_begin +  out2.trianglelist[base+2]-1).base() 	)
+            );
+
+
+#ifdef _DEBUG
+            ModelPart::NodesContainerType& ModelNodes = ThisModelPart.Nodes();
+            if( *(ModelNodes).find( out2.trianglelist[base]).base() == *(ThisModelPart.Nodes().end()).base() )
+                KRATOS_THROW_ERROR(std::logic_error,"trying to use an inexisting node","");
```

Comment:
Here the macro is correct but changing it to new format would give you more information about the error.